_posts/2016-04-26-aws-and-consul.md

```diff
@@ -0,0 +1,175 @@
+---
+layout:     post
+title:      Deploying Consul in AWS 
+date:       2016-04-26 07:00:00
+summary:    Looking at deploying a Consul cluster into AWS and how we can rollout upgrades with no loss of service
+author:     adam_pointer
+image:      consul_logo.png
+category:   Deployment
+tags: consul, aws, autoscaling, terraform
+---
+
+Before we start talking about how deploying Consul into AWS, it probably makes sense to start by giving a brief overview of what Consul actually is. For those of you who do not know what Consul is at all, it is a distributed service discovery and configuration application. Think *Apache Zookeeper* or *etcd*, but written by those clever chaps at HashiCorp. For this post, I really want to focus more on Consul's clustering capabilities, as this is what really sets it apart and enables us to do some neat tricks with AWS to allow rolling updates with no loss of service and also how we can make a cluster totally self-healing.
```

Comment:
Shouldn't the opening be "Before we start talking about how we deploy Consul into AWS, ..."?